README.md

```diff
@@ -17,3 +17,120 @@ with Zipkin-compatible data model.
+    <artifactId>jaeger-jaxrs2</artifactId>
+    <version>0.5.0</version>
+</dependency>
+```
+
+### Core ###
+
+If you only want to do custom instrumentation using the core tracing functionality then add the following:
+```
+<dependency>
+    <groupId>com.uber.jaeger</groupId>
+    <artifactId>jaeger-core</artifactId>
+    <version>0.5.0</version>
+</dependency>
+```
+
+## Configuration: ##
+
+The minimal configuration in base.yaml should look like this:
+```
+jaeger:
+  serviceName: geocache
+```
+In a yaml configuration file you must specify a `serviceName`.  It is also possible to specify a ‘disable’ flag set to true, or false.  Jaeger should ALWAYS be ENABLED except in case of emergencies where you are sure there is a problem with Jaeger.  If the `disable` field is left out then Jaeger will be enabled by default.
+
+
+## JAX-RS Instrumentation: ##
+
+As of right now we have provided instrumentation for everything that follows the JAX-RS 2.0 standard.  JAX-RS based implementations such as Jersey and Dropwizard allow you to specify client and server filters to do some computational work before every request or response.  Thus to add Jaeger to your service you just need to have a server filter set on your server, and a client filter set on you outbound client requests.  Note it is important for all Java services to have a central place to configure Clients, for outgoing requests, as well as a central place for servers.
+
+### Dropwizard: ###
+
+Jaeger’s dropwizard jar exposes a configuration object `com.uber.jaeger.dropwizard.Configuration`  If your service reads in its configuration through a pojo object then you can use this Configuration class to read the yaml format specified above.
+The following is an example of adding a client filter to a JAX-RS based framework that creates a dropwizard/jersey client.
+
+Client instrumentation example:
+```
+import com.uber.jaeger.dropwizard.TracingUtils;
+
+// JaegerConfig gets initialized by whichever dropwizard configuration reader you are using.
+com.uber.jaeger.dropwizard.Configuration jaegerConfig;
+
+final Client client = new JerseyClientBuilder(env)
+    /* Arbitrary configuration code … */
+    .register(TracingUtils.clientFilter(jaegerConfig);
+```
+
+Your server configuration will probably not look exactly like the one below.  However JAX-RS server configurations always expose a `register` function that can be used to register a server filter like below.
+Server instrumentation example:
+```
+Final ResourceConfig rc = new ResourceConfig()
+    /* Arbitrary configuration code… */
+    .register(TracingUtils.serverFilter(jaegerConfig));
+
+return GrizzlyHttpServerFactory.createHttpServer(URI.create(BASE_URI), rc);
+```
+
+### Metrics ###
+In order for Jaeger to set metrics properly it needs to be configured with a metric registry.  
+```
+import com.uber.jaeger.dropwizard.TracingUtils;
+
+// Dropwizard's metrics registry class
+MetricsRegistry registry = new MetricsRegistry();
```

Comment:
this will not compile. Let's give a better example. Better yet, we should have a sample application - #24